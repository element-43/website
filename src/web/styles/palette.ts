export default {
    brand: {
        // Primary.
        purple300: '#9a6dea',
        purple400: '#8154d1',
        purple500: '#673ab7',
        purple600: '#4e219e',
        purple700: '#340784',
        purple800: '#1b006b',

        // Accent.
        green300: '#61ffa4',
        green400: '#48e68b',
        green500: '#2ecc71',
        green600: '#15b358',
        green700: '#00993e',
        green800: '#008025',
    },

    greyScale: {
        white: '#fff',
        lightGrey: '#f2f2f2',
        grey: '#ccc',
        darkGrey: '#8c8c8c',
        black: '#000',
    },

    ui: {
        darkRed: '#8b0000',
        orange: '#ffa500',
        orangeRed: '#ff4500',
        pink: '##ff69b4',
        red: '#cc0000',
        salmon: '#fa8072',
        slateGrey: '#708090',
        yellow: '#ffff00',
    },
};
