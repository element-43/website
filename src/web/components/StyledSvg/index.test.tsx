import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';

// Config.
import palette from '../../styles/palette';

// Components.
import { Props, StyledSvg } from './index';

interface Scope {
  props: Props;
  wrapper: ShallowWrapper;
}

describe('/components/StyledSvg', () => {
  let scope: Scope;

  beforeEach(() => {
    const props: Props = {
      viewBox: '0 0 0 0',
    };

    scope = {
      props,
      wrapper: shallow(<StyledSvg {...props} />),
    };
  });

  describe('<SvgIcon /> snapshots', () => {
    it('should match the snapshot with the default props', () => {
      expect(scope.wrapper).toMatchSnapshot();
    });

    it('should match the snapshot with the optional props', () => {
      let wrapper: ShallowWrapper;

      scope.props.color = palette.ui.red;
      scope.props.hoverColor = palette.greyScale.white;
      scope.props.size = '100px';

      wrapper = shallow(<StyledSvg {...scope.props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
