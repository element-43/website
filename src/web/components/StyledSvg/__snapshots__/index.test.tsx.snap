// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/StyledSvg <SvgIcon /> snapshots should match the snapshot with the default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.svg
    viewBox="0 0 0 0"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bdVaJa dUNsyo",
      "viewBox": "0 0 0 0",
    },
    "ref": null,
    "rendered": null,
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bdVaJa dUNsyo",
        "viewBox": "0 0 0 0",
      },
      "ref": null,
      "rendered": null,
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`/components/StyledSvg <SvgIcon /> snapshots should match the snapshot with the optional props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.svg
    color="#cc0000"
    hoverColor="#fff"
    size="100px"
    viewBox="0 0 0 0"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bdVaJa iqZbym",
      "color": "#cc0000",
      "size": "100px",
      "viewBox": "0 0 0 0",
    },
    "ref": null,
    "rendered": null,
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bdVaJa iqZbym",
        "color": "#cc0000",
        "size": "100px",
        "viewBox": "0 0 0 0",
      },
      "ref": null,
      "rendered": null,
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
